presto-pinot-toolkit/src/main/java/com/facebook/presto/pinot/query/PinotFilterExpressionConverter.java

```diff
@@ -201,6 +202,24 @@ else if (timeFieldExpression instanceof ConstantExpression) {
         return Optional.of(timeValueString);
     }
 
+    private PinotExpression handleContains(
+            CallExpression contains,
+            Function<VariableReferenceExpression, Selection> context)
+    {
+        if (contains.getArguments().size() == 2) {
+            RowExpression left = contains.getArguments().get(0);
+            RowExpression right = contains.getArguments().get(1);
+            Preconditions.checkState(right instanceof ConstantExpression, format("Contains operator can not push down non-literal value: %s", right));
```

Comment:
How about a simple Pinot unsupported exception here ?

Or is this a bug in presto ? Like does presto (over hive) allow a non literal here ? If it allows and we merely want to forbid pushdown then, Pinot connector should use unsupported exception here.

Reply:
Not only literal, it can also be and identifier, e.g. a column, so I can check if this array field contains another single value column value in the same row.

Reply:
updated the exception